x/leverage/keeper/token.go

```diff
@@ -7,12 +7,54 @@ import (
 	"github.com/umee-network/umee/v3/x/leverage/types"
 )
 
-// SetTokenSettings stores a Token into the x/leverage module's KVStore.
-func (k Keeper) SetTokenSettings(ctx sdk.Context, token types.Token) error {
-	if err := token.Validate(); err != nil {
-		return err
+// UpdateTokenSettings will update the token in registry or save the new token to registry
+// or remove the tokens from registry.
+func (k Keeper) UpdateTokenSettings(ctx sdk.Context, updateRegistryTokens []types.Token) error {
+	registryTokens := k.GetAllRegisteredTokens(ctx)
+	registryTokensBaseDenoms := make(map[string]types.Token)
+	for _, token := range registryTokens {
+		registryTokensBaseDenoms[token.BaseDenom] = token
+	}
+
+	updateRegistryTokenDenoms := make(map[string]types.Token)
+	for _, token := range updateRegistryTokens {
+		updateRegistryTokenDenoms[token.BaseDenom] = token
+	}
+
+	for baseDenom := range registryTokensBaseDenoms {
+		if _, ok := updateRegistryTokenDenoms[baseDenom]; ok {
+			// update the token in registry
+			if err := k.SetTokenSettings(ctx, updateRegistryTokenDenoms[baseDenom]); err != nil {
+				return err
+			}
+
+			k.Logger(ctx).Info("updated the token settings", "base_denom", baseDenom)
+		} else {
+			// remove the token settings
+			if err := k.RemoveTokenSettings(ctx, registryTokensBaseDenoms[baseDenom]); err != nil {
+				return err
+			}
+
+			k.Logger(ctx).Info("removed the token settings", "base_denom", baseDenom)
+		}
+
+		delete(updateRegistryTokenDenoms, baseDenom)
```

Comment:
This should be simplified:

- Do not create a map
- For every token in proposal, `SetTokenSettings`
- Make no distinction between adding and updating existing tokens
- Never remove any token